File: python/min-max.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  min-max.py

#Random.randint
import random 


def losuj (ile,lmin, lmax):
    lista = [] 
    for i in range(ile):
        lista.append(random.randint(lmin, lmax))
    print(lista)
    return lista
    
    
def minmax():
    ile = int(input('Ile podasz liczb?'))
    min = max = int(input('Podaj liczbę: '))
    for i in range(ile - 1):
        liczba = int(input('Podaj liczbę: '))
        if min > liczba:
            min = liczba
        if max < liczba:
           max = liczba 
    return min, max
    
    
def minimum(liczby):
    najm = liczby[0]
    for i in liczby:
        if najm > i:
            najm = i
    return najm        
            
def maximum(liczby):
    najw = liczby[0]
    for i in liczby:
        if najw < i:
            najw = i
    return najw        
    
    
    
    
    
    
def main(args):
    # ~min, max = minmax()
    # ~print('Najmniejsza:;' , min) 
    # ~print('Najwieksza:;', max)
    liczby = losuj(20, 0, 50)
    print(' Najmniejsza: ', minimum(liczby)) 
    print(' Najwieksza: ', maximum(liczby)) 
    
    
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/minmax.cxx
/*
 * minmax.cxx

 */


#include <iostream>

using namespace std;

void wypelnij(int tab[], int roz) {
    cout << "Wprowadź rozmiar" << roz << "liczb: " << endl;
    for(int i=0; i<roz; i++) {
        cin >> tab[i];
        
        } 
    
}

void drukuj(int tab[], int roz) {
    for(int i=0; i<roz; i++) {
        cout << tab[i] << " " << endl;
        
        } 
    
}

void minmax(int tab[], int roz) {
    int minimum = tab[0];
    int maximum = tab[0];
    for(int i=1; i<roz; i++) {
        if (minimum > tab[i])
            minimum = tab[i];
        if (maximum < tab[i])
            maximum = tab[i]; 
            

        }
        
    cout << "liczba min =" << minimum << endl;
    cout << "liczba max =" << maximum << endl; 
    
}



int main(int argc, char **argv)
{
    const int rozmiar = 5;
    
    int tab[rozmiar];
    wypelnij (tab, rozmiar);
	drukuj (tab, rozmiar);
    cout << endl;
    minmax (tab, rozmiar);
	return 0;
}

